### ISSUE TITLE
High voltage on VCC rail (discussion).

### ISSUE THREAD

**Issue by joksan** [ghidraninja/game-and-watch-base#5]
I'm bringing the #1 discussion here so as not to derail it from the original topic.

**Comment by joksan** [comment]
Thanks for the oscillograms. I assume those are shots from the VCC rails at different moments. Seems like I should program the MCU and test how it behaves upon toggling the different power domain control lines.

**Comment by kbeckmann** [comment]
Ah thanks for creating a new issue. Yes, indeed. They were from the VCC rail during powerup when VUSB was applied.

If you are going to experiment with this i can recommend you to desolder the flash in order to protect it from high voltage. The MCU is 3.3v capable and it can handle up to 4V without getting damaged (table 17 https://www.st.com/resource/en/datasheet/stm32h7b0ib.pdf ), but higher than that will probably not be very good for it.

**Comment by joksan** [comment]
I'm already experimenting and very probably putting my unit into risk. Although no damage seems to have occurred despite triggering the high voltage issue so many times.

I'm quite perplexed by the mode of failure. Manually toggling the domain enable lines doesn't seem to trigger the issue.

From what I have gathered pin 85 enables 3.3V on the LCD (active high), while pin 82 toggles power on the exteral flash and another portion of the LCD (active low). Pin 85 seems to drive what seems to be the enable line on a 3.3V regulator (backlight?) and pin 82 drives what seems to be a mosfet that simply passes 1.8V to flash and quite possibly the LCD control logic.

By the way, that colored GIF is really helpful. Thanks to Upcycle-Electronics for such valuable info.

**Comment by joksan** [comment]
What I have gathered so far...

- When triggered, the rail on the output side of the 1.9V stays at about 4V even after the power control domain lines are turned off.
- In order to clear the problem, the entire board needs to be power cycled. Turning on and off the device doesn't clear it.
- The MOSFET simply passes 4V to flash and LCD. It doesn't really seem to be a problem related with it.

**Comment by joksan** [comment]
This is a bit scary, but causing a very brief short circuit to ground on the output of the 1.9V regulator, clears the error condition and causes voltage to drop down back to 1.9V. This is hinting a positive loop situation somewhere. Could the MCU be involved in delivering 4V through one of it pins to where it shouldn't?

**Comment by joksan** [comment]
![image](https://user-images.githubusercontent.com/1202639/101297143-fe1c2900-37ec-11eb-8d6d-a87222255493.jpeg)
I had absolutely forgotten I had this little gadget in my desk.

The regulator is as I suspected a switching one. A buck type to be precise. The signal you're seeing is the switch output (on the other side of the inductor, the one that isn't Vcc).

**Comment by joksan** [comment]
And... the moment of truth. This is what happens when we trigger the failure:
![image](https://user-images.githubusercontent.com/1202639/101297307-f5782280-37ed-11eb-8378-35251a7bec88.jpeg)
That is, an steady 4V output from the switch.

**Comment by kbeckmann** [comment]
Oh! That's super interesting. Almost looks like the feedback is broken somehow? I mean the buck should have a feedback line that is used to balance the output voltage.

**Comment by joksan** [comment]
The regulator is quite simply dying on us. The output switch gets stuck at the ON condition. Now, there is a feedback pin, but that is already giving the 4V back to the regulator.

**Comment by joksan** [comment]
This could be a failure on the design of the regulator chip itself. Probably triggered by the MCU triggering loads too fast. The regulator is quite possibly entering some kind of internal deadlock condition.

**Comment by joksan** [comment]
Now, the battery could be somehow dampening the effect of loads being triggered. I'm going to bet that placing a capacitor along the input terminals will fix the issue.

**Comment by joksan** [comment]
![image](https://user-images.githubusercontent.com/1202639/101297631-84396f00-37ef-11eb-9040-b6cdf997265b.jpeg)
Yup! Absolutely as expected. I just added an aluminum capacitor by some precarious pin bending. Turned the unit on and off over 30 times in a row. No metaestable condition was triggered.

**Comment by joksan** [comment]
Now, there's already a multi-layer capacitor on the board (In fact I put the aluminum one in parallel by touching their pins together), but seems like it's not enough for dampening the inrush current caused by loads being turned on.

This absolutely expains why, by adding delays, you managed to solve the issue in software, as you give the circuit some time to recover from the voltage drops.

Too bad that the part number that Nintendo chose has this issue. If this were other kind of design, where output voltage is absolutely critical, this could have fried tons of units.

At least, I think we can now put this issue to rest now. Or at least for me, my OCD is already satisfied ;-)

**Comment by kbeckmann** [comment]
Woah. Thanks for your thorough investigation. So what you're saying is, that the sleep is actually the proper solution here without a hardware mod?

**Comment by joksan** [comment]
Yes! absolutely!

**Comment by kbeckmann** [comment]
Cool. Then I can also sleep well after this.

**Comment by joksan** [comment]
From now on, I'll try not to turn the unit whitout the battery inside. We should probably recommend this to everyone, as the battery being present prevents this from happening.

**Comment by joksan** [comment]
Of course being careful in software has an added benefit. But as we know, not all programs take this extra precaution, in particular, the stock firmware doesn't.

**Comment by kbeckmann** [comment]
Thanks again for your investigation. I am closing this issue as the problem has been fully understood now.

**Comment by joksan** [comment]
![image](https://user-images.githubusercontent.com/1202639/102843720-6cf1a880-43cf-11eb-9113-ab2e7ac5542e.png)
Sorry for necromancing a topic that's already closed, but I think I can get a little more insight into this topic. We've been having some talk with discord user \_tim\_ about this issue, looking for ways to avoid this issue through software only. Here's an oscilloscope capture at the instant of the failure. In yellow we have the PMC output (the input to the 1.8V regulator) and in Magenta we have the output of the 1.8V regulator. Voltage scales are 1V/div for both channels and 100us/div for time. As we can see, the PMC output drops quite dramatically, causing the 1.8V regulator to fail and latch the output transistor permanently on.

**Comment by joksan** [comment]
![image](https://user-images.githubusercontent.com/1202639/102844185-7deee980-43d0-11eb-97f5-8d109f5d3b9c.png)
If, in the other hand, we leave the battery on, we can still see a drop, although this time the drop is quite minor and doesn't cause the 1.8V regulator to even budge a little, proving with almost certainty that leaving the battery on causes the voltage drop to be dampened. This behavior is for the stock firmware by the way.

**Comment by joksan** [comment]
The length of the drop for my unit is quite short (about 400uS or 0.4ms), meaning that the safe time marging should be above this value. Though I'm quite certain that there will be variations from device from device, as this is highly dependant on the internal resistance/impedance of the PMC. Though further tests are still required using homebrew firmware, as I'm not certain which voltage domain is causing trouble (could be either the secondary 1.8V domain or 3.3V).

I'm leaving this here as reference, so others can quickly check. Have a nice day!

**Comment by kbeckmann** [comment]
Thanks for the updates!

**Comment by mat231228** [comment]
Hi all,
The battery charger IC has an in-rush current limit feature on VIN (VUSB) designed to meet USB guidelines. Current limit is set to approx. 200mA by pins EN1 (logic low), EN2 (logic high) and ILIM resistor (7.5kOhms) (see [datasheet ](https://www.ti.com/lit/ds/slus810m/slus810m.pdf) and [schematic](https://github.com/Upcycle-Electronics/game-and-watch-hardware/blob/main/Images-Version-Current/GnWschematic1v2.pdf)).

With the battery connected, any supplementary current over these 200mA drawn by the application circuit (for example in-rush current when turning LCD backlight on) will be drawn from the battery.

In the absence of a battery, the current limit kicks in and the storage caps on bus V3v7 get depleted, causing a voltage drop and the behaviour observed with the 1.8V buck converter (EDIT: which might be an [SCR latch-up](https://en.m.wikipedia.org/wiki/Latch-up) caused by the sudden power supply negative spike). Of course adding bulk storage helps, as experienced by @joksan above.

Changing R17 to 1.1kOhms would set the current limiter to the chip's maximum supported 1.5A and might also help. Of course, that would probably break strict USB compliance (maybe?).

I guess Nintendo never intended the board to function properly without a battery :-). 

All of this is just a guess from reading the schematic and datasheet, I have not yet opened up my Game and Watch unit. Let me know if it makes sense.
